examples/bri-3/src/bri/zeroKnowledgeProof/services/circuit/circuitInputsParser/circuitInputParser.service.spec.ts

```diff
@@ -0,0 +1,259 @@
+          dataType: 'integer',
+        } as CircuitInputMapping,
+      ],
+    } as CircuitInputsMapping;
+
+    // Act
+    const circuitInputs = cips.applyMappingToJSONPayload(payload, schema);
+
+    // Assert
+    expect(circuitInputs).toStrictEqual({ dascircuitinput: 123 });
+  });
+
+  it('Should generate a single circuit input param based on a default value for a missing integer param at root level', () => {
+    // Arrange
+    const payload = `{
+        "somethingElse": ""
+       }`;
+
+    const defaultValue = 555333;
+
+    const schema = {
+      mapping: [
+        {
+          circuitInput: 'dascircuitinput',
+          description: 'desc',
+          payloadJsonPath: 'supplierInvoiceID',
+          dataType: 'integer',
+          defaultValue: defaultValue,
+        } as CircuitInputMapping,
+      ],
+    } as CircuitInputsMapping;
+
+    // Act
+    const circuitInputs = cips.applyMappingToJSONPayload(payload, schema);
+
+    // Assert
+    expect(circuitInputs).toStrictEqual({ dascircuitinput: defaultValue });
+  });
+
+  it('Should return null based on a missing default value for a missing integer param at root level', () => {
+    // Arrange
+    const payload = `{
+        "somethingElse": ""
+       }`;
+
+    const schema = {
+      mapping: [
+        {
+          circuitInput: 'dascircuitinput',
+          description: 'desc',
+          payloadJsonPath: 'supplierInvoiceID',
+          dataType: 'integer',
+        } as CircuitInputMapping,
+      ],
+    } as CircuitInputsMapping;
+
+    // Act
+    const circuitInputs = cips.applyMappingToJSONPayload(payload, schema);
+
+    // Assert
+    expect(circuitInputs).toBeNull();
+  });
+
+  it('Should generate a single circuit input param based on a single integer array param at root level', () => {
+    // Arrange
+    const payload = `{
+        "supplierInvoiceIDs": [
+          123,
+          321,
+          454
+        ]
+       }`;
+
+    const schema = {
+      mapping: [
+        {
+          circuitInput: 'dascircuitinput',
+          description: 'desc',
+          payloadJsonPath: 'supplierInvoiceIDs',
+          dataType: 'array',
+          arrayType: 'integer',
+        } as CircuitInputMapping,
+      ],
+    } as CircuitInputsMapping;
+
+    // Act
+    const circuitInputs = cips.applyMappingToJSONPayload(payload, schema);
+
+    // Assert
+    expect(circuitInputs).toStrictEqual({ dascircuitinput: [123, 321, 454] });
+  });
+
+  it('Should generate a single circuit input param with charcode sums based on a single string array param at root level', () => {
+    // Arrange
+    const payload = `{
+        "supplierInvoiceIDs": [
+          "INV123",
+          "INV124",
+          "INV125"
+        ]
+       }`;
+
+    const schema = {
+      mapping: [
+        {
+          circuitInput: 'dascircuitinput',
+          description: 'desc',
+          payloadJsonPath: 'supplierInvoiceIDs',
+          dataType: 'array',
+          arrayType: 'string',
```

Comment:
do we need both array and arrayType? can arrayType be calculated using `typeof array[0]`? can `typeof value` be used in other places too to simplify mapping logic and remove some fields here?

i feel more validation of this object is needed, for example if dataType is not array, we should not specifiy arrayType and so on, but it seems like some simplification can be done first

Reply:
We want the participants to strictly define the types that are expected in the payload, so that we can do a type check before we even try to invoke the circuit. there is still definitely room for improvement, both for simplification and additional validation, but let's do it in small iterations as we move forward.

Reply:
agree. the `circuitInput` for example will be an object that has a schema itself depending on the type of input parameters.